pkg/kv/kvserver/replica_proposal_buf.go

```diff
@@ -700,9 +705,19 @@ func (b *propBuf) maybeRejectUnsafeProposalLocked(
 			return false
 		}
 		ownsCurrentLease := b.p.ownsValidLease(ctx, b.clock.NowAsClockTimestamp())
-		if li.leaderEligibleForLease && !ownsCurrentLease && !b.testing.allowLeaseProposalWhenNotLeader {
+
+		reject := false
+		if !li.leaderKnown() && RejectLeaseOnLeaderUnknown.Get(&b.settings.SV) {
+			log.VEventf(ctx, 2, "not proposing lease acquisition because we're not the leader; the leader is unknown")
+			reject = true
+		} else if li.leaderEligibleForLease && !ownsCurrentLease && !b.testing.allowLeaseProposalWhenNotLeader {
```

Comment:
[pre-existing] The `allowLeaseProposalWhenNotLeader` name sounds more related to the no-leader case above, should we move it up?

Reply:
Added a TODO.